Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -3914,7 +3915,7 @@ void WebPageProxy::receivedNavigationPolicyDecision(PolicyAction policyAction, A
         return;
     }
 
-    Ref<WebProcessProxy> sourceProcess = process();
+    Ref<WebProcessProxy> sourceProcess = sourceProcessForNavigation;
```

Comment:
I think this change is leading to failure in `SiteIsolation.ChildNavigatingToNewDomain`:

```
ASSERTION FAILED: !m_messageReceivers.contains(std::make_pair(messageReceiverName, destinationID))
/Volumes/Data/safari-webkit/OpenSource/Source/WebKit/Platform/IPC/MessageReceiverMap.cpp(53) : void IPC::MessageReceiverMap::addMessageReceiver(ReceiverName, uint64_t, MessageReceiver &)
1   0x10df13d08 WTFCrash
2   0x12c138814 WTFCrashWithInfo(int, char const*, char const*, int)
3   0x12f14fc70 IPC::MessageReceiverMap::addMessageReceiver(IPC::ReceiverName, unsigned long long, IPC::MessageReceiver&)
4   0x12dcafca0 WebKit::AuxiliaryProcessProxy::addMessageReceiver(IPC::ReceiverName, unsigned long long, IPC::MessageReceiver&)
5   0x12d7bb1bc WebKit::AuxiliaryProcessProxy::addMessageReceiver(IPC::ReceiverName, WTF::ObjectIdentifierGenericBase const&, IPC::MessageReceiver&)
6   0x12dd6093c WebKit::SubframePageProxy::SubframePageProxy(WebKit::WebPageProxy&, WebKit::WebProcessProxy&, bool)
7   0x12dd60b48 WebKit::SubframePageProxy::SubframePageProxy(WebKit::WebPageProxy&, WebKit::WebProcessProxy&, bool)
8   0x12dfb7be0 WTF::UniqueRef<WebKit::SubframePageProxy> WTF::makeUniqueRefWithoutFastMallocCheck<WebKit::SubframePageProxy, WebKit::WebPageProxy&, WTF::Ref<WebKit::WebProcessProxy, WTF::RawPtrTraits<WebKit::WebProcessProxy>>&, bool>(WebKit::WebPageProxy&, WTF::Ref<WebKit::WebProcessProxy, WTF::RawPtrTraits<WebKit::WebProcessProxy>>&, bool&&)
```